core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponseParser.java

```diff
@@ -76,17 +76,20 @@ public static ErrorResponse fromJson(JsonNode jsonNode) {
         jsonNode != null && jsonNode.isObject(),
         "Cannot parse error response from non-object value: %s",
         jsonNode);
-    Preconditions.checkArgument(jsonNode.has(ERROR), "Cannot parse missing field: error");
-    JsonNode error = JsonUtil.get(ERROR, jsonNode);
-    String message = JsonUtil.getStringOrNull(MESSAGE, error);
-    String type = JsonUtil.getStringOrNull(TYPE, error);
-    Integer code = JsonUtil.getIntOrNull(CODE, error);
-    List<String> stack = JsonUtil.getStringListOrNull(STACK, error);
-    return ErrorResponse.builder()
-        .withMessage(message)
-        .withType(type)
-        .responseCode(code)
-        .withStackTrace(stack)
-        .build();
+    if (jsonNode.has(ERROR)) {
+      JsonNode error = JsonUtil.get(ERROR, jsonNode);
+      String message = JsonUtil.getStringOrNull(MESSAGE, error);
+      String type = JsonUtil.getStringOrNull(TYPE, error);
+      Integer code = JsonUtil.getIntOrNull(CODE, error);
+      List<String> stack = JsonUtil.getStringListOrNull(STACK, error);
+      return ErrorResponse.builder()
+          .withMessage(message)
+          .withType(type)
+          .responseCode(code)
+          .withStackTrace(stack)
+          .build();
+    } else {
+      return ErrorResponse.builder().build();
+    }
```

Comment:
I maybe missing something but this is `ErrorResponseParser` no?  I'd expect whatever JSON that gets passed to this to have the error. I just assumed the response model isn't marked as required in the REST spec, because it depends on if an error is thrown as part of the call. If it is thrown, I'd expect all these fields to be set. Maybe there's a better way to convey it in the spec.

Reply:
We can also update the spec, but it looks like not all systems (EMR) are sending the full message. The question is, do we want to fail at parsing the error message, or just return an empty message (or throw an exception somewhere else).